src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyCultureInfo.cs

```diff
@@ -41,43 +41,48 @@ public override void Initialize(AnalysisContext analysisContext)
 
             analysisContext.RegisterCompilationStartAction(csaContext =>
             {
-                var obsoleteAttributeType = csaContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemObsoleteAttribute);
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(csaContext.Compilation);
```

Comment:
Nit: Can we avoid pulling in `wellKnownTypeProvider` into the analyzers? I don't believe saving an additional `WellKnownTypeProvider.GetOrCreate` for each well known type gives any real performance benefit, and ends up unnecessarily having the analyzer deal with the provider. I think prior implementation has better readability.

Reply:
Fine by me! Do you want to always avoid using the `WellKnownTypeProvider`?

Reply:
Lets avoid it until we really need it (say like an argument passed to some existing API being invoked). If we find this is indeed a performance bottleneck, we can likely do a repo wide change to have analyzers use WellKnownTypeProvider explicitly when accessing more than one well known type.